integration/inspektor-gadget/run_schedcls_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 The Inspektor Gadget authors
```

Comment:
Do you have a specific idea of how this test could be implement as a unit one?

Reply:
With image-based gadgets Inspektor Gadget is becoming more a framework than a collection of tools, hence we need to make sure the framework is working fine. I think we should be testing the framework part (the ability to pull, load,  attach, confiture, etc.. ebpf programs.) without using real gadgets. The reasons are:
- It's possible that don't have gadgets that use all features of the framework. The TC support on this PR is one example of that.
- It's easier to understand if an issue is on the framework rather than in the gadget.
- Tests are easier to implement as we don't depend on the gadget's logic.
- By using mock gadgets we could be able to test more corner cases than by using real ones.
- It can happen that we'll move the "official gadgets" to an independent repository later on, so using them to test will add an "external dependency"

I agree we need more unit tests, but these ones are needed as well.

https://github.com/inspektor-gadget/inspektor-gadget/issues/2301 is following this discussion.